resources/prosody-plugins/mod_muc_allowners.lua

```diff
@@ -1,12 +1,99 @@
+local jid = require "util.jid";
 local is_healthcheck_room = module:require "util".is_healthcheck_room;
 
+local moderated_subdomains;
+local moderated_rooms;
+
+local function load_config()
+    moderated_subdomains = module:get_option_set("allowners_moderated_subdomains", {})
+    moderated_rooms = module:get_option_set("allowners_moderated_rooms", {})
+end
+load_config();
+
+-- Checks whether the jid is moderated, the room name is in moderated_rooms
+-- or if the subdomain is in the moderated_subdomains
+-- @return returns on of the:
+--      -> false
+--      -> true, subdomain, nil (in case of matching by subdomain)
+--      -> true, nil, room_name (if matched by room name)
+local function isModerated(room_jid)
+    local room_node = jid.node(room_jid);
+    -- parses bare room address, for multidomain expected format is:
+    -- [subdomain]roomName@conference.domain
+    local target_subdomain = room_node:match("^%[([^%]]+)%](.+)$");
+
+    if target_subdomain then
+        if moderated_subdomains:contains(target_subdomain) then
+            return true, target_subdomain, nil;
+        end
+    elseif moderated_rooms:contains(room_node) then
+        return true, nil, room_node;
+    end
+
+    return false;
+end
+
+-- we need to hook on stanzas to be able to prevent focus(jicofo) from setting owner for the
```

Comment:
Is there still a valid use-case for jicofo doing that? Even if there is, can we make that configurable instead?